src/vc/explain.rs

```diff
@@ -164,6 +167,20 @@ pub(super) fn explain_annotated_while(
                 {
                     return explain_park_induction(vcgen, &args[0], body);
                 }
+
+                if anno_ref
+                    .as_any()
+                    .downcast_ref::<UnrollAnnotation>()
+                    .is_some()
+                {
+                    let enc_env = EncodingEnvironment {
```

Comment:
Could you move the `enc_env` creation into the `explain_unroll` fn and add add a comment indicating what information in this struct is "dummy" and what is not?